### PR TITLE
Added a custom server example using Flask

### PR DESCRIPTION
**Pull request by jrast**
See issue https://github.com/mirumee/ariadne/issues/125

### CODE REVIEW

**Comment by jrast** [comment]
Thanks for the review @tbsf 
I will update the PR accordingly.

**Comment by jrast** [comment]
Ok, I updated the example. It's working now with Ariadne 0.3.0 and includes the recomendations of @tbsf

**Comment by rafalp** [comment]
I'm wondering if this shouldn't be 3rd party library, eg. `flask_ariadne` with readme, tests and Pypi release.

I'm not strongly opinionated but I'm worrying that we may be setting an expectation to the folk that our docs will eventually cover all Python web frameworks, and that's not going to work.

**Comment by tbsf** [comment]
This is just my experience, but finding a Flask example was a deciding factor in giving this a try instead of sticking with Apollo. I didn't even notice there was also documentation for Django. Regardless of the route you choose to go down for documentation, I definitely suggest calling more attention to the ability to create a custom servers and mention that Django and Flask are supported in the README itself.

See Apollo's list of integrations for example: https://github.com/apollographql/apollo-server#installation-integrations

**Comment by rafalp** [comment]
@tbsf this is excellent feedback, I haven't considered this. Thanks!

**Comment by jrast** [comment]
I don't see the need for a 3rd party library at the moment. It's only about 20 lines of code which are needed and a package for these lines seems to be an overkill and would remind me of some npm packages.

On the other side, I agree that it will be impossible to have a example for each web framework. And if a lot of framework specific examples are in the docs, issues about the framework integration will pop up which are difficult to solve if the maintainers are not working with these frameworks.
Maybe a note can be added in the docs that these examples are more a "Proof of Concept" and no support will be given?

**Comment by rafalp** [comment]
I'll do the pragmatic thing here and assume I've been trying to optimize prematurely ;)

@jrast can you please update the PR to rename "custom servers" to "Integrations" and rename "Basic GraphQL server with ..." to "... example"? I know @patrys also wants to follow up with "Starlette example".

File: docs/custom-server.rst [hunk 1]
```diff
@@ -102,3 +102,66 @@ The following example presents a basic GraphQL server using a Django framework::
         status_code = 200 if success else 400
         # Send response to client
         return JsonResponse(result, status=status_code)
+
+
+Basic GraphQL server with Flask
```
Comment (patrys) [inline docs/custom-server.rst]:
```suggestion
Integrating with Flask
```

"Basic" could be taken to mean "that's not suitable for real-world use".

Comment (rafalp) [reply docs/custom-server.rst]: Starting chapters with framework name will make for better copy  and navigation IMHO:

```
- Django integration
- Flask integration
- Starlette integration
```

**Comment by rafalp** [comment]
@jrast thank you for help! I'll close this PR and open my own that includes `Flask` chapter you've written as well some more generic pointers for other integration authors.

**Pull request closed without merging**